rules/dead-code/src/Rector/FunctionLike/RemoveOverriddenValuesRector.php

```diff
@@ -175,27 +175,28 @@ private function resolveNodesToRemove(array $assignedVariableNames, array $nodes
         foreach ($assignedVariableNames as $assignedVariableName) {
             $previousNode = null;
 
-            foreach ($nodesByTypeAndPosition as $nodeByTypeAndPosition) {
-                $variableNode = $nodeByTypeAndPosition->getVariableNode();
+            foreach ($nodesByTypeAndPosition as $nodes) {
+                $variableNode = $nodes->getVariableNode();
```

Comment:
@TomasVotruba I think we need to check the static type of the variable before use of previous `By` usage, as `$nodesByTypeAndPosition`, the `$nodes` is a single Values object, not array.

Reply:
@TomasVotruba I tried to check the value var of foreach is Array Type, but it always got Mixed Type at https://github.com/rectorphp/rector/pull/5696/commits/adb9f767f20ea2544e5ccfc9cfe361ecc929912a , do you have idea how to get it? Thank you.

Reply:
I reverted the type of value var of the foreach, so use as `nodes` as is.